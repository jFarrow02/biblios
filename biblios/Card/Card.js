"use strict";Object.defineProperty(exports,"__esModule",{value:true});exports["default"]=void 0;var _react=_interopRequireDefault(require("react"));var _propTypes=_interopRequireDefault(require("prop-types"));var _styledComponents=_interopRequireDefault(require("styled-components"));var _styles=_interopRequireDefault(require("../styles"));function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{"default":obj}}var StyledCard=_styledComponents["default"].div.withConfig({displayName:"Card__StyledCard",componentId:"sc-1gijcbc-0"})(["margin-top:10px;margin-bottom:10px;box-shadow:3px 3px 5px 3px rgba(102,102,102,.5);max-width:","px;display:flex;flex-direction:column;"],function(props){return props.width?props.width:500});var StyledBottomBorder=_styledComponents["default"].div.withConfig({displayName:"Card__StyledBottomBorder",componentId:"sc-1gijcbc-1"})(["background-color:",";width:95%;height:2px;margin:0 auto;"],function(props){return props.footerColor?props.footerColor:_styles["default"].gray03});var StyledContentHeader=_styledComponents["default"].div.withConfig({displayName:"Card__StyledContentHeader",componentId:"sc-1gijcbc-2"})(["padding:20px 10px;"]);var StyledCardFooter=_styledComponents["default"].div.withConfig({displayName:"Card__StyledCardFooter",componentId:"sc-1gijcbc-3"})(["background-color:",";padding:20px 10px;"],function(props){return props.footerColor?props.footerColor:_styles["default"].gray03});var StyledCardContent=_styledComponents["default"].div.withConfig({displayName:"Card__StyledCardContent",componentId:"sc-1gijcbc-4"})(["padding:20px 10px;"]);var Card=function Card(props){var content=props.content,footer=props.footer,footerContent=props.footerContent,header=props.header,headerContent=props.headerContent;return _react["default"].createElement(StyledCard,null,header&&_react["default"].createElement(_react["default"].Fragment,null,_react["default"].createElement(StyledContentHeader,null,headerContent),_react["default"].createElement(StyledBottomBorder,null)),_react["default"].createElement(StyledCardContent,null,content),footer&&_react["default"].createElement(StyledCardFooter,null,footerContent))};var _default=Card;exports["default"]=_default;Card.propTypes={color:_propTypes["default"].string,content:_propTypes["default"].oneOfType([_propTypes["default"].element,_propTypes["default"].string]),footer:_propTypes["default"].bool,footerColor:_propTypes["default"].string,footerContent:_propTypes["default"].oneOfType([_propTypes["default"].element,_propTypes["default"].string]),header:_propTypes["default"].bool,headerContent:_propTypes["default"].oneOfType([_propTypes["default"].element,_propTypes["default"].string]),width:_propTypes["default"].PropTypes.number};Card.defaultProps={color:null,content:"",footer:false,footerColor:null,footerContent:"",header:false,headerContent:""};