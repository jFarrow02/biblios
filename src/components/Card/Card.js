import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import stylesConfig from '../styles';

const StyledCard = styled.div`
        margin-top: 10px;
        margin-bottom: 10px;
        box-shadow: 3px 3px 5px 3px rgba(204, 204, 204, .5);
        max-width: ${props => props.width ? props.width : 500}px;
        display: flex;
        flex-direction: column;
        h1, h2, h3, h4, h5, h6{
            margin: 0px;
            margin-block-start: 0px;
            margin-block-end: 0px;
            padding: 10px;
        }
    `;

    const StyledContentHeader = styled.div`
        border-bottom: 1px solid ${stylesConfig.gray03};
    `;
    
    const StyledCardFooter = styled.div`
        background-color: ${stylesConfig.gray03};
    `;

    const StyledCardContent = styled.div`
        padding: 20px 10px;
    `;

const Card = props =>{

    const {content, footer, footerContent, header, headerContent} = props;
    
    return(
       <StyledCard>
           {
            header&&
           <StyledContentHeader>{headerContent}</StyledContentHeader>
           }
           <StyledCardContent>
               {content}
           </StyledCardContent>
           {
            footer&& 
            <StyledCardFooter>
                {footerContent}
            </StyledCardFooter>
           }
       </StyledCard>
    );
}

export default Card;

Card.propTypes = {
    content: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
    footer: PropTypes.bool,
    footerContent: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
    header: PropTypes.bool,
    headerContent: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
    width: PropTypes.PropTypes.number,
};

Card.defaultProps = {
    content: '',
    footer: false,
    footerContent: '',
    header: false,
    headerContent: '',
};