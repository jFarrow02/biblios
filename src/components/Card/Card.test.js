import React from 'react';
import Card from './Card';
import renderer from 'react-test-renderer';

it('should match a snapshot', ()=>{
    const tree = renderer.create(<Card/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});