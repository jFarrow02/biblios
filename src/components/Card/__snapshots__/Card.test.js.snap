// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match a snapshot 1`] = `
<div
  className="Card__StyledCard-sc-1gijcbc-0 laoIgD"
>
  <div
    className="Card__StyledCardContent-sc-1gijcbc-3 dMkYTu"
  >
    
  </div>
</div>
`;
